lib/logger/lib/logger/formatter.ex

```diff
@@ -94,19 +102,21 @@ defmodule Logger.Formatter do
   defp compile_code(key) when key in @valid_patterns, do: key
 
   defp compile_code(key) when is_atom(key) do
-    raise ArgumentError, "$#{key} is an invalid format pattern."
+    raise ArgumentError, "$#{key} is an invalid format pattern"
   end
 
   @doc """
   Formats time as chardata.
   """
+  @spec format_time({0..23, 0..59, 0..59, 0..999}) :: IO.chardata()
```

Comment:
This (and the spec for `format_time/1` are not optimal IMO. We should extract them to a `time/0` type and a `date/0` type, and have `format/5` take a `{date(), time()}` type, but we would change the meaning of the `time/0` type. This would be a breaking change I guess, that's why I didn't do it.

Reply:
Hmm... Could the seconds be 60 in case of a leap second or is it ignored in this format like in posix?

Reply:
I just copied the type from `t:Logger.Formatter.time/0`, which is what is essentially broken down and passed to these two formatting functions.